Description: Fill in mips Makefile rules
 Upstream is about to delete the Makefiles in the next version, so this will
 remain a Debian-specific patch.
Author: Ximin Luo <infinity0@debian.org>
Bug: https://github.com/rust-lang/rust/pull/38650
Forwarded: not-needed
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/mk/cfg/mips-unknown-linux-gnu.mk
+++ b/mk/cfg/mips-unknown-linux-gnu.mk
@@ -18,7 +18,7 @@
 CFG_WINDOWSY_mips-unknown-linux-gnu :=
 CFG_UNIXY_mips-unknown-linux-gnu := 1
 CFG_LDPATH_mips-unknown-linux-gnu :=
-CFG_RUN_mips-unknown-linux-gnu=
-CFG_RUN_TARG_mips-unknown-linux-gnu=
+CFG_RUN_mips-unknown-linux-gnu=$(2)
+CFG_RUN_TARG_mips-unknown-linux-gnu=$(call CFG_RUN_mips-unknown-linux-gnu,,$(2))
 RUSTC_FLAGS_mips-unknown-linux-gnu :=
 CFG_GNU_TRIPLE_mips-unknown-linux-gnu := mips-unknown-linux-gnu
--- a/mk/cfg/mips64-unknown-linux-gnuabi64.mk
+++ b/mk/cfg/mips64-unknown-linux-gnuabi64.mk
@@ -1 +1,24 @@
-# rustbuild-only target
+# mips64-unknown-linux-gnuabi64 configuration
+CC_mips64-unknown-linux-gnuabi64=mips64-linux-gnuabi64-gcc
+CXX_mips64-unknown-linux-gnuabi64=mips64-linux-gnuabi64-g++
+CPP_mips64-unknown-linux-gnuabi64=mips64-linux-gnuabi64-gcc
+AR_mips64-unknown-linux-gnuabi64=mips64-linux-gnuabi64-ar
+CFG_LIB_NAME_mips64-unknown-linux-gnuabi64=lib$(1).so
+CFG_STATIC_LIB_NAME_mips64-unknown-linux-gnuabi64=lib$(1).a
+CFG_LIB_GLOB_mips64-unknown-linux-gnuabi64=lib$(1)-*.so
+CFG_LIB_DSYM_GLOB_mips64-unknown-linux-gnuabi64=lib$(1)-*.dylib.dSYM
+CFG_JEMALLOC_CFLAGS_mips64-unknown-linux-gnuabi64 := -mips64r2 -mabi=64 $(CFLAGS)
+CFG_GCCISH_CFLAGS_mips64-unknown-linux-gnuabi64 := -Wall -g -fPIC -mips64r2 -mabi=64 $(CFLAGS)
+CFG_GCCISH_CXXFLAGS_mips64-unknown-linux-gnuabi64 := -fno-rtti $(CXXFLAGS)
+CFG_GCCISH_LINK_FLAGS_mips64-unknown-linux-gnuabi64 := -shared -fPIC -g -mips64r2
+CFG_GCCISH_DEF_FLAG_mips64-unknown-linux-gnuabi64 := -Wl,--export-dynamic,--dynamic-list=
+CFG_LLC_FLAGS_mips64-unknown-linux-gnuabi64 :=
+CFG_INSTALL_NAME_mips64-unknown-linux-gnuabi64 =
+CFG_EXE_SUFFIX_mips64-unknown-linux-gnuabi64 :=
+CFG_WINDOWSY_mips64-unknown-linux-gnuabi64 :=
+CFG_UNIXY_mips64-unknown-linux-gnuabi64 := 1
+CFG_LDPATH_mips64-unknown-linux-gnuabi64 :=
+CFG_RUN_mips64-unknown-linux-gnuabi64=$(2)
+CFG_RUN_TARG_mips64-unknown-linux-gnuabi64=$(call CFG_RUN_mips64-unknown-linux-gnuabi64,,$(2))
+RUSTC_FLAGS_mips64-unknown-linux-gnuabi64 :=
+CFG_GNU_TRIPLE_mips64-unknown-linux-gnuabi64 := mips64-unknown-linux-gnuabi64
--- a/mk/cfg/mips64el-unknown-linux-gnuabi64.mk
+++ b/mk/cfg/mips64el-unknown-linux-gnuabi64.mk
@@ -1 +1,24 @@
-# rustbuild-only target
+# mips64el-unknown-linux-gnuabi64 configuration
+CC_mips64el-unknown-linux-gnuabi64=mips64el-linux-gnuabi64-gcc
+CXX_mips64el-unknown-linux-gnuabi64=mips64el-linux-gnuabi64-g++
+CPP_mips64el-unknown-linux-gnuabi64=mips64el-linux-gnuabi64-gcc
+AR_mips64el-unknown-linux-gnuabi64=mips64el-linux-gnuabi64-ar
+CFG_LIB_NAME_mips64el-unknown-linux-gnuabi64=lib$(1).so
+CFG_STATIC_LIB_NAME_mips64el-unknown-linux-gnuabi64=lib$(1).a
+CFG_LIB_GLOB_mips64el-unknown-linux-gnuabi64=lib$(1)-*.so
+CFG_LIB_DSYM_GLOB_mips64el-unknown-linux-gnuabi64=lib$(1)-*.dylib.dSYM
+CFG_JEMALLOC_CFLAGS_mips64el-unknown-linux-gnuabi64 := -mips64r2 -mabi=64 $(CFLAGS)
+CFG_GCCISH_CFLAGS_mips64el-unknown-linux-gnuabi64 := -Wall -g -fPIC -mips64r2 -mabi=64 $(CFLAGS)
+CFG_GCCISH_CXXFLAGS_mips64el-unknown-linux-gnuabi64 := -fno-rtti $(CXXFLAGS)
+CFG_GCCISH_LINK_FLAGS_mips64el-unknown-linux-gnuabi64 := -shared -fPIC -g -mips64r2
+CFG_GCCISH_DEF_FLAG_mips64el-unknown-linux-gnuabi64 := -Wl,--export-dynamic,--dynamic-list=
+CFG_LLC_FLAGS_mips64el-unknown-linux-gnuabi64 :=
+CFG_INSTALL_NAME_mips64el-unknown-linux-gnuabi64 =
+CFG_EXE_SUFFIX_mips64el-unknown-linux-gnuabi64 :=
+CFG_WINDOWSY_mips64el-unknown-linux-gnuabi64 :=
+CFG_UNIXY_mips64el-unknown-linux-gnuabi64 := 1
+CFG_LDPATH_mips64el-unknown-linux-gnuabi64 :=
+CFG_RUN_mips64el-unknown-linux-gnuabi64=$(2)
+CFG_RUN_TARG_mips64el-unknown-linux-gnuabi64=$(call CFG_RUN_mips64el-unknown-linux-gnuabi64,,$(2))
+RUSTC_FLAGS_mips64el-unknown-linux-gnuabi64 :=
+CFG_GNU_TRIPLE_mips64el-unknown-linux-gnuabi64 := mips64el-unknown-linux-gnuabi64
--- a/mk/cfg/mipsel-unknown-linux-gnu.mk
+++ b/mk/cfg/mipsel-unknown-linux-gnu.mk
@@ -1,7 +1,7 @@
 # mipsel-unknown-linux-gnu configuration
 CC_mipsel-unknown-linux-gnu=mipsel-linux-gnu-gcc
 CXX_mipsel-unknown-linux-gnu=mipsel-linux-gnu-g++
-CPP_mipsel-unknown-linux-gnu=mipsel-linux-gnu-gcc
+CPP_mipsel-unknown-linux-gnu=mipsel-linux-gnu-gcc -E
 AR_mipsel-unknown-linux-gnu=mipsel-linux-gnu-ar
 CFG_LIB_NAME_mipsel-unknown-linux-gnu=lib$(1).so
 CFG_STATIC_LIB_NAME_mipsel-unknown-linux-gnu=lib$(1).a
@@ -18,7 +18,7 @@
 CFG_WINDOWSY_mipsel-unknown-linux-gnu :=
 CFG_UNIXY_mipsel-unknown-linux-gnu := 1
 CFG_LDPATH_mipsel-unknown-linux-gnu :=
-CFG_RUN_mipsel-unknown-linux-gnu=
-CFG_RUN_TARG_mipsel-unknown-linux-gnu=
+CFG_RUN_mipsel-unknown-linux-gnu=$(2)
+CFG_RUN_TARG_mipsel-unknown-linux-gnu=$(call CFG_RUN_mipsel-unknown-linux-gnu,,$(2))
 RUSTC_FLAGS_mipsel-unknown-linux-gnu :=
 CFG_GNU_TRIPLE_mipsel-unknown-linux-gnu := mipsel-unknown-linux-gnu
